src/storage/redis_db.cc

```diff
@@ -937,7 +937,7 @@ rocksdb::Status Database::Sort(const RedisType &type, const std::string &key, co
     for (const std::string &pattern : args.getpatterns) {
       std::string val = lookupKeyByPattern(pattern, elem.obj);
       if (val.empty()) {
-        elems->emplace_back(redis::NilString(version));
+        elems->emplace_back("");
```

Comment:
Hmmm seems we need to distinguish between nil and empty string here?

If so we need to use something like `vector<optional<string>>`.

Reply:
I agree with you, this also requires modifying `lookupKeyByPattern` and adding relevant tests